core/src/layers/prometheus.rs

```diff
@@ -0,0 +1,990 @@
+
+    // batch
+    pub batch_request_counts: GenericCounter<AtomicU64>,
+    pub batch_request_latency: Histogram,
+
+    // blocking create
+    pub blocking_create_request_counts: GenericCounter<AtomicU64>,
+    pub blocking_create_request_latency: Histogram,
+
+    // blocking read
+    pub blocking_read_request_counts: GenericCounter<AtomicU64>,
+    pub blocking_read_request_latency: Histogram,
+    pub blocking_read_size: Histogram,
+
+    // blocking write
+    pub blocking_write_request_counts: GenericCounter<AtomicU64>,
+    pub blocking_write_request_latency: Histogram,
+    pub blocking_write_size: Histogram,
+
+    // blocking stat
+    pub blocking_stat_request_counts: GenericCounter<AtomicU64>,
+    pub blocking_stat_request_latency: Histogram,
+
+    // blocking delete
+    pub blocking_delete_request_counts: GenericCounter<AtomicU64>,
+    pub blocking_delete_request_latency: Histogram,
+
+    // blocking list
+    pub blocking_list_request_counts: GenericCounter<AtomicU64>,
+    pub blocking_list_request_latency: Histogram,
+
+    // blocking scan
+    pub blocking_scan_request_counts: GenericCounter<AtomicU64>,
+    pub blocking_scan_request_latency: Histogram,
+}
+
+impl PrometheusMetrics {
+    /// new with prometheus register.
+    pub fn new(registry: Registry) -> Self {
+        // metadata
+        let metadata_request_counts = register_int_counter_with_registry!(
+            "metadata_request_counts",
+            "Total times of metadata be called",
+            registry
+        )
+        .unwrap();
+        let opts = histogram_opts!(
+            "metadata_request_latency",
+            "Histogram of the time spent on getting metadata",
+            exponential_buckets(0.01, 2.0, 16).unwrap()
+        );
+
+        let metadata_request_latency = register_histogram_with_registry!(opts, registry).unwrap();
+
+        // create
+        let create_request_counts = register_int_counter_with_registry!(
+            "create_request_counts",
+            "Total times of create be called",
+            registry
+        )
+        .unwrap();
+        let opts = histogram_opts!(
+            "create_request_latency",
+            "Histogram of the time spent on creating",
+            exponential_buckets(0.01, 2.0, 16).unwrap()
+        );
+
+        let create_request_latency = register_histogram_with_registry!(opts, registry).unwrap();
+
+        // read
+        let read_request_counts = register_int_counter_with_registry!(
+            "read_request_counts",
+            "Total times of read be called",
+            registry
+        )
+        .unwrap();
+        let opts = histogram_opts!(
+            "read_request_latency",
+            "Histogram of the time spent on reading",
+            exponential_buckets(0.01, 2.0, 16).unwrap()
+        );
+        let read_request_latency = register_histogram_with_registry!(opts, registry).unwrap();
+
+        let opts = histogram_opts!(
```

Comment:
We used `Counter` to store size and users can calculate the other metrics by dividing `duration_seconds` or sum up to calculate the total size.

Reply:
From my personal perspective, I prefer to use `HistogramVec`(`Histogram`) to monitor size, for the reason, you can refer to the pr description.